nvme.c

```diff
@@ -5317,6 +5321,330 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
 	return err;
 }
 
+bool nvme_is_fabrics_reg(int offset)
+{
+	switch (offset) {
+	case NVME_REG_CAP:
+		fallthrough;
+	case NVME_REG_VS:
+		fallthrough;
+	case NVME_REG_CC:
+		fallthrough;
+	case NVME_REG_CSTS:
+		fallthrough;
+	case NVME_REG_NSSR:
+		fallthrough;
+	case NVME_REG_CRTO:
+		return true;
+	default:
+		break;
+	}
+
+	return false;
+}
+
+static int get_register_property(int fd, void **pbar, int offset)
+{
+	int err, size = getpagesize();
+	__u64 value;
+	void *bar = malloc(size);
+	struct nvme_get_property_args args = {
+		.args_size	= sizeof(args),
+		.fd		= fd,
+		.offset		= offset,
+		.value		= &value,
+		.timeout	= NVME_DEFAULT_IOCTL_TIMEOUT,
+	};
+
+	if (!bar) {
+		nvme_show_error("malloc: %s", strerror(errno));
+		return -1;
+	}
+
+	memset(bar, 0xff, size);
```

Comment:
Also seems like we should be able to get rid of this large `memset()` since only the property is going to be used

Reply:
I see so just fixed the size to reduce by the commit cc597115 from the page size to the register offset to get.

Reply:
Okay, just allocating for the fabric registers seems small enough, thanks